ocaml/typing/global_module.mli

```diff
@@ -1,18 +1,27 @@
 [@@@ocaml.warning "+a-9-40-41-42"]
 
+type ('name, 'value) duplicate =
+  | Duplicate of { name : 'name; value1 : 'value; value2 : 'value }
+
+module Argument : sig
+  type ('param, 'value) t = {
+    param : 'param;
+    value : 'value;
+  }
+end
+
 module Name : sig
   type t = private {
     head : string;
     args : argument list;
   }
-  and argument = {
-    param : t;
-    value : t;
-  }
+  and argument = (t, t) Argument.t
 
   include Identifiable.S with type t := t
 
-  val create : string -> argument list -> t
+  val create : string -> argument list -> (t, (t, t) duplicate) Result.t
```

Comment:
It seems that we either trust the build system to generate well-formed instances, or `args` is empty. Then we don't need to expose this function?

Reply:
I'd rather not bake in the assumption that the build system is writing these. It's still source code and CLI arguments, which we generally treat with nice error messages. (We do have the awful syntax on the assumption that the build system is writing it, but that's much easier to change later.)

The no-argument case is frequent enough to make a separate function, though. Adding `create_no_args : string -> t`. (I might have called it `of_string` but it's not actually the inverse of `to_string`.)

Reply:
I like `create_no_args`. Resolving.